File: 2299-strong-password-checker-ii/2299-strong-password-checker-ii.cpp
class Solution {
public:
    bool strongPasswordCheckerII(string p) {
        if(size(p) < 8) return false;
        bool low = false , upper = false ,digit = false ,special = false;
        for(auto c : p){
            if(c>='a' and c<='z') low = true;
            else if(c>='A' and c <='Z') upper = true;
            else if(isdigit(c)) digit = true;
            else special = true;
        }
        //Check the 6th condition
        for(int i=0;i+1<size(p);i++) if(p[i] == p[i+1]) return false;
        if(low and upper and digit and special) return true;
        return false;
    }
};

File: 2299-strong-password-checker-ii/2299-strong-password-checker-ii.java
class Solution {
    public boolean strongPasswordCheckerII(String password) {
       char[] ch = password.toCharArray();               // original string
        char[] specials = ("!@#$%^&*()-+").toCharArray(); // special characters
        int n = ch.length;
        if(n < 8) return false; 
        boolean u = false, l = false, d = false, s = false;
        for(int i = 1; i < n; i++) if(ch[i] == ch[i - 1]) return false; // adjacent characters should not be same
        for(char c : ch) {
            u |= (c >= 'A' && c <= 'Z');
            d |= (c >= '0' && c <= '9');
            l |= (c >= 'a' && c <= 'z');
            for(char ss : specials) s |= (ss == c);
        }
        return (u && l && s && d);   
    }
}